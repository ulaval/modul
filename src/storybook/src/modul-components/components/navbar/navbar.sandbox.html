<div>
    <!-- Sandbox template -->
    <h2>Skins</h2>
    <p class="m-u--margin-bottom">
        This page shows the possible skins for the navbar component.
        You can use the navbar like an navigation menu or like tabs. Both have a main level and sub level skins, combine them to create beautiful two levels menus or tabs.
    </p>

    <m-button @click="toggleMultiline">{{ multiline ? 'Remove' : 'Add' }} multiline</m-button>

    <h2>Navigation menu</h2>
    <h3>Fisrt level</h3>
    <p>Skin nav-main (default)</p>
    <m-navbar :selected.sync="selectedItem"
              :multiline="true"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1"
                       label="1 souris verte"
                       :use-new-wrap-strategy="true">
        </m-navbar-item>
        <m-navbar-item value="item2"
                       label="2"
                       :use-new-wrap-strategy="true">
            2
        </m-navbar-item>
        <!-- <m-navbar-item value="item8"
                       label="8 asdasda asd asd asd"
                       :use-new-wrap-strategy="true">
        </m-navbar-item> -->
        <m-navbar-item value="item9"
                       label="9 singes"
                       :use-new-wrap-strategy="true">
        </m-navbar-item>
        <m-navbar-item value="item12"
                       label="12 zaine d oeufs"
                       :use-new-wrap-strategy="true">
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true"
                       label="10 chevres et 10 moutons"
                       :use-new-wrap-strategy="true">
        </m-navbar-item>
        <m-navbar-item value="item11"
                       label="11  volant vers le lointain vraiment"
                       :use-new-wrap-strategy="true">
        </m-navbar-item>
    </m-navbar>

    <!-- <p>Skin nav-soft</p>
    <m-navbar skin="nav-soft"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant"
              style="background: #09f;">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufs
        </m-navbar-item>
    </m-navbar>

    <h3>Sub level</h3>
    <p>Skin nav-sub</p>
    <m-navbar skin="nav-sub"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufssss
        </m-navbar-item>
    </m-navbar> -->

    <!-- <h3>Two levels</h3>
    <p>nav-main and nav-sub combined</p>
    <m-navbar skin="nav-main"
              :selected.sync="selectedItemMainMenu"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1"
                       @click="toggleNavPres">
            Présentation du cours
        </m-navbar-item>
        <m-navbar-item value="item2"
                       @click="toggleNavMod">
            Modules d'apprentissage
        </m-navbar-item>
        <m-navbar-item value="item3"
                       @click="closeNav">
            Évaluations et travaux pratiques
        </m-navbar-item>
        <m-navbar-item value="item4"
                       @click="closeNav">
            Résultats
        </m-navbar-item>
        <m-navbar-item value="item5"
                       @click="closeNav">
            Ressources complémentaires et références bibliographiques
        </m-navbar-item>
        <m-navbar-item value="item6"
                       @click="closeNav">
            Calendrier de session
        </m-navbar-item>
    </m-navbar>
    <m-accordion-transition>
        <div v-if="navOpenPres">
            <m-navbar skin="nav-sub"
                      :selected.sync="selectedItemPres">
                <m-navbar-item value="item1">
                    Introduction
                </m-navbar-item>
                <m-navbar-item value="item2">
                    Liens entre le cours et le programme
                </m-navbar-item>
                <m-navbar-item value="item3">
                    Modalités particulières de l’offre de cours
                </m-navbar-item>
                <m-navbar-item value="item4">
                    Objectifs généraux
                </m-navbar-item>
                <m-navbar-item value="item5">
                    Objectifs particuliers
                </m-navbar-item>
                <m-navbar-item value="item6">
                    Éléments de contenu
                </m-navbar-item>
                <m-navbar-item value="item7">
                    Approche pédagogique
                </m-navbar-item>
                <m-navbar-item value="item8">
                    Modalités d'encadrement
                </m-navbar-item>
                <m-navbar-item value="item9">
                    Appréciation de l'enseignement
                </m-navbar-item>
            </m-navbar>
        </div>
    </m-accordion-transition>
    <m-accordion-transition>
        <div v-if="navOpenMod">
            <m-navbar skin="nav-sub"
                      :selected.sync="selectedItemMod">
                <m-navbar-item value="item1">
                    Module de la semaine 1
                </m-navbar-item>
                <m-navbar-item value="item2">
                    Module de la semaine 2
                </m-navbar-item>
                <m-navbar-item value="item3">
                    Module de la semaine 3
                </m-navbar-item>
                <m-navbar-item value="item4">
                    Module de la semaine 4
                </m-navbar-item>
                <m-navbar-item value="item5">
                    Module de la semaine 5
                </m-navbar-item>
                <m-navbar-item value="item6">
                    Module de la semaine 6
                </m-navbar-item>
                <m-navbar-item value="item7">
                    Module de la semaine 7
                </m-navbar-item>
                <m-navbar-item value="item8">
                    Module de la semaine 8
                </m-navbar-item>
                <m-navbar-item value="item9">
                    Module de la semaine 9
                </m-navbar-item>
            </m-navbar>
        </div>
    </m-accordion-transition>

    <h2>Tabs</h2>
    <h3>Fist level</h3>
    <p>Skin tab-light-main : only one tab</p>
    <m-navbar skin="tab-light"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
    </m-navbar>

    <p>Skin tab-light-main : more than one tab</p>
    <m-navbar skin="tab-light-main"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufs
        </m-navbar-item>
    </m-navbar>

    <p>Skin tab-dark-main : only one tab</p>
    <m-navbar skin="tab-dark-main"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
    </m-navbar>

    <p>Skin tab-dark-main : more than one tab</p>
    <m-navbar skin="tab-dark-main"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufs
        </m-navbar-item>
    </m-navbar>

    <h3>Sub level</h3>
    <p>Skin tab-arrow</p>
    <m-navbar skin="tab-arrow"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufs
        </m-navbar-item>
    </m-navbar>

    <p>Skin tab-underline</p>
    <m-navbar skin="tab-underline"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufs
        </m-navbar-item>
    </m-navbar>

    <p>Skin tab-soft</p>
    <m-navbar skin="tab-soft"
              :selected.sync="selectedItem"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1">
            1 souris verte
        </m-navbar-item>
        <m-navbar-item value="item2">
            2
        </m-navbar-item>
        <m-navbar-item value="item3">
            3
        </m-navbar-item>
        <m-navbar-item value="item4">
            4
        </m-navbar-item>
        <m-navbar-item value="item5">
            5
        </m-navbar-item>
        <m-navbar-item value="item6">
            6
        </m-navbar-item>
        <m-navbar-item value="item7">
            7
        </m-navbar-item>
        <m-navbar-item value="item8">
            8
        </m-navbar-item>
        <m-navbar-item value="item9">
            9 singes
        </m-navbar-item>
        <m-navbar-item value="item10"
                       :disabled="true">
            10 chèvres et 10 moutons
        </m-navbar-item>
        <m-navbar-item value="item11">
            11 barnacles magestueuses volant vers le lointain
        </m-navbar-item>
        <m-navbar-item value="item12">
            12 zaine d'oeufs
        </m-navbar-item>
    </m-navbar>

    <h3>Two levels</h3>
    <p>nav-main and nav-sub combined</p>
    <m-navbar skin="tab-light-main"
              :selected.sync="selectedItemMainMenu"
              :multiline="multiline"
              title-button-left="Naviguer précédent"
              title-button-right="Naviguer suivant">
        <m-navbar-item value="item1"
                       @click="toggleNavPres">
            Présentation du cours
        </m-navbar-item>
        <m-navbar-item value="item2"
                       @click="toggleNavMod">
            Modules d'apprentissage
        </m-navbar-item>
        <m-navbar-item value="item3"
                       @click="closeNav">
            Évaluations et travaux pratiques
        </m-navbar-item>
        <m-navbar-item value="item4"
                       @click="closeNav">
            Résultats
        </m-navbar-item>
        <m-navbar-item value="item5"
                       @click="closeNav">
            Ressources complémentaires et références bibliographiques
        </m-navbar-item>
        <m-navbar-item value="item6"
                       @click="closeNav">
            Calendrier de session
        </m-navbar-item>
    </m-navbar>
    <m-accordion-transition>
        <div v-if="navOpenPres">
            <m-navbar skin="tab-underline"
                      :selected.sync="selectedItemPres">
                <m-navbar-item value="item1">
                    Introduction
                </m-navbar-item>
                <m-navbar-item value="item2">
                    Liens entre le cours et le programme
                </m-navbar-item>
                <m-navbar-item value="item3">
                    Modalités particulières de l’offre de cours
                </m-navbar-item>
                <m-navbar-item value="item4">
                    Objectifs généraux
                </m-navbar-item>
                <m-navbar-item value="item5">
                    Objectifs particuliers
                </m-navbar-item>
                <m-navbar-item value="item6">
                    Éléments de contenu
                </m-navbar-item>
                <m-navbar-item value="item7">
                    Approche pédagogique
                </m-navbar-item>
                <m-navbar-item value="item8">
                    Modalités d'encadrement
                </m-navbar-item>
                <m-navbar-item value="item9">
                    Appréciation de l'enseignement
                </m-navbar-item>
            </m-navbar>
        </div>
    </m-accordion-transition>
    <m-accordion-transition>
        <div v-if="navOpenMod">
            <m-navbar skin="tab-underline"
                      :selected.sync="selectedItemMod">
                <m-navbar-item value="item1">
                    Module de la semaine 1
                </m-navbar-item>
                <m-navbar-item value="item2">
                    Module de la semaine 2
                </m-navbar-item>
                <m-navbar-item value="item3">
                    Module de la semaine 3
                </m-navbar-item>
                <m-navbar-item value="item4">
                    Module de la semaine 4
                </m-navbar-item>
                <m-navbar-item value="item5">
                    Module de la semaine 5
                </m-navbar-item>
                <m-navbar-item value="item6">
                    Module de la semaine 6
                </m-navbar-item>
                <m-navbar-item value="item7">
                    Module de la semaine 7
                </m-navbar-item>
                <m-navbar-item value="item8">
                    Module de la semaine 8
                </m-navbar-item>
                <m-navbar-item value="item9">
                    Module de la semaine 9
                </m-navbar-item>
            </m-navbar>
        </div>
    </m-accordion-transition>

    <h2>No skin</h2>
    <p>Navbar with no style applied</p>
    <div style="background-color: black; color: pink;">
        <m-navbar skin="plain"
                  :selected.sync="selectedItem"
                  :multiline="multiline"
                  title-button-left="Naviguer précédent"
                  title-button-right="Naviguer suivant">
            <m-navbar-item value="item1">
                1 souris verte
            </m-navbar-item>
            <m-navbar-item value="item2">
                2
            </m-navbar-item>
            <m-navbar-item value="item3">
                3
            </m-navbar-item>
            <m-navbar-item value="item4">
                4
            </m-navbar-item>
            <m-navbar-item value="item5">
                5
            </m-navbar-item>
            <m-navbar-item value="item6">
                6
            </m-navbar-item>
            <m-navbar-item value="item7">
                7
            </m-navbar-item>
            <m-navbar-item value="item8">
                8
            </m-navbar-item>
            <m-navbar-item value="item9">
                9 singes
            </m-navbar-item>
            <m-navbar-item value="item10"
                           :disabled="true">
                10 chèvres et 10 moutons
            </m-navbar-item>
            <m-navbar-item value="item11">
                11 barnacles magestueuses volant vers le lointain
            </m-navbar-item>
            <m-navbar-item value="item12">
                12 zaine d'oeufs
            </m-navbar-item>
        </m-navbar>
    </div> -->

    <!-- <h2>Support for vue-router named route (MODUL-616)</h2>
    <m-navbar :selected.sync="selectedItemMod">
        <m-navbar-item value="item1">
        </m-navbar-item>
        <m-navbar-item value="item2"
                       :url="{path: '/'}">
        </m-navbar-item>
    </m-navbar> -->
</div>
