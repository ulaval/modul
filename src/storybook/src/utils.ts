export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
