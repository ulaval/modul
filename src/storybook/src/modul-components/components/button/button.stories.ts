import { actions } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { BUTTON_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';


export default {
    title: `${modulComponentsHierarchyRootSeparator}${BUTTON_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    props: {
        text: {
            default: text('Text', 'A Button')
        }
    },
    methods: actions(
        'click',
        'focus',
        'blur'
    ),
    template: '<m-button @click="click" @focus="focus" @blur="blur">{{ text }}</m-button>'
});

defaultStory.story = {
    name: 'default'
};

export const disabled = () => '<m-button :disabled="true">A Button</m-button>';

export const waiting = () => '<m-button :waiting="true">A Button</m-button>';

export const fullsize = () => '<m-button fullSize="true">A Button</m-button>';

export const precision = () => '<m-button >A Button <template slot="precision">Button precision</template></m-button>';

export const submit = () => '<m-button type="submit">A Button</m-button>';

export const reset = () => '<m-button type="reset">A Button</m-button>';

export const secondaryDefault = () => '<m-button skin="secondary">A Button</m-button>';

export const secondaryDisabled = () => '<m-button skin="secondary" :disabled="true">A Button</m-button>';

export const secondaryWaiting = () => '<m-button skin="secondary" :waiting="true">A Button</m-button>';

export const secondaryFullSize = () => '<m-button :fullSize="true" skin="secondary">A Button</m-button>';

export const secondaryIcon = () => '<m-button icon-name="m-svg__close-clear" skin="secondary">A Button</m-button>';

export const secondaryIcon20Px = () => '<m-button icon-name="m-svg__close-clear" icon-size="20px" skin="secondary">A Button</m-button>';

export const iconPositionRight = () => '<m-button icon-name="m-svg__close-clear" icon-position="right" skin="secondary">A Button</m-button>';

export const secondaryPrecision = () => '<m-button skin="secondary" >A Button <template slot="precision">Button precision</template></m-button>';

export const icon = () => '<m-button icon-name="m-svg__close-clear">A Button</m-button>';

export const iconDisabled = () => '<m-button :disabled="true" icon-name="m-svg__close-clear">A Button</m-button>';

export const iconWaiting = () => '<m-button icon-name="m-svg__close-clear" waiting="true">A Button</m-button>';

export const iconFullSize = () => '<m-button :fullSize="true" icon-name="m-svg__close-clear">A Button</m-button>';

export const icon20Px = () => '<m-button icon-name="m-svg__close-clear" icon-size="20px">A Button</m-button>';

export const iconPrecision = () => '<m-button icon-name="m-svg__close-clear">A Button <template slot="precision">' +
    'Button precision</template></m-button>';
