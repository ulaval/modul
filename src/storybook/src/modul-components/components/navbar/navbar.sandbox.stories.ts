import { NAVBAR_NAME } from '@ulaval/modul-components/dist/components/component-names';

import { sandboxDecorator } from '../../../sandbox-decorator';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${NAVBAR_NAME}`
};

export const sandbox: any = () => '<m-navbar-sandbox></m-navbar-sandbox>';

sandbox.story = {
    decorators: [sandboxDecorator],
    parameters: { fileName: __filename, options: { showPanel: false, isToolshown: true } }
};
