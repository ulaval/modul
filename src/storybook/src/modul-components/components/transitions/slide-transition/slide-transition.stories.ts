import { storiesOf } from '@storybook/vue';
import { MButtonSkin } from '@ulaval/modul-components/dist/components/button/button';
import { SLIDE_TRANSITION_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MNavbarSkin } from '@ulaval/modul-components/dist/components/navbar/navbar';
import SlideTransitionPlugin, { MSlideTransitionDirection } from '@ulaval/modul-components/dist/components/transitions/slide-transition/slide-transition';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../../utils';


Vue.use(SlideTransitionPlugin);

storiesOf(`${modulComponentsHierarchyRootSeparator}${SLIDE_TRANSITION_NAME}`, module)

    .add('Tabs example', () => ({
        data: () => ({
            selectedTabIndex: 1,
            direction: MSlideTransitionDirection.LeftToRight,
            nbParagraphe: 20,
            nbTab: 7,
            skinNavbar: MNavbarSkin.TabLight,
            disabledTransition: false,
            buttonSkin: MButtonSkin.Secondary,
            transitionEmit: ''
        }),
        methods: {
            showTab(newIndex): void {
                let _this: any = this as any;
                _this.direction = (newIndex < _this.selectedTabIndex) ? MSlideTransitionDirection.LeftToRight : MSlideTransitionDirection.RightToLeft;
                _this.selectedTabIndex = newIndex;
                _this.nbParagraphe = (_this.selectedTabIndex % 2) === 0 ? 10 : newIndex * 10;
            },
            beforeEnter(): void {
                (this as any).transitionEmit = 'before-enter';
                console.log(`@emit('before-enter')`);
            },
            enter(): void {
                (this as any).transitionEmit = 'enter';
                console.log(`@emit('enter')`);
            },
            afterEnter(): void {
                (this as any).transitionEmit = 'after-enter';
                console.log(`@emit('after-enter')`);
            },
            enterCancelled(): void {
                (this as any).transitionEmit = 'enter-cancelled';
                console.log(`@emit('enter-cancelled')`);
            },
            beforeLeave(): void {
                (this as any).transitionEmit = 'before-leave';
                console.log(`@emit('before-leave')`);
            },
            leave(): void {
                (this as any).transitionEmit = 'leave';
                console.log(`@emit('leave')`);
            },
            afterLeave(): void {
                (this as any).transitionEmit = 'after-leave';
                console.log(`@emit('after-leave')`);
            },
            leaveCancelled(): void {
                (this as any).transitionEmit = 'leave-cancelled';
                console.log(`@emit('leave-cancelled')`);
            }
        },
        template: `
        <div>
            <m-navbar :selected="selectedTabIndex"
                      :skin="skinNavbar">
                <m-navbar-item v-for="index in nbTab"
                               :key="index"
                               :value="index"
                               @click="showTab(index)">
                    Tab {{index}}
                </m-navbar-item>
            </m-navbar>
            <m-slide-transition :direction="direction"
                                :disabled="disabledTransition"
                                @before-enter="beforeEnter"
                                @enter="enter"
                                @after-enter="afterEnter"
                                @enter-cancelled="enterCancelled"
                                @before-leave="beforeLeave"
                                @leave="leave"
                                @after-leave="afterLeave"
                                @leave-cancelled="leaveCancelled">
                <div v-for="index in nbTab"
                     :key="index"
                     v-if="selectedTabIndex === index"
                     class="m-u--margin-top">
                    <h3 class="m-u--no-margin">Tab {{index}}</h3>
                    <p v-for="index in nbParagraphe"
                       :key="index">Lorem ipsum dolor sit amet, consectetur adipisicing elit. A laudantium odio ipsum, quae quos cum dolor, omnis perferendis, veritatis deleniti iusto consectetur? Impedit tempora quam ab laborum maiores sapiente earum?</p>
                </div>
            </m-slide-transition>
            <div class="m-u--margin-top--m">
                <m-button @click="nbTab++"
                          :disabled="nbTab === 12">Add tab</m-button>
                <m-button :skin="buttonSkin"
                          class="m-u--margin-left"
                          :disabled="nbTab === 2"
                          @click="nbTab--">Delete tab</m-button>
                <m-button :skin="buttonSkin"
                          class="m-u--margin-left"
                          @click="disabledTransition = !disabledTransition">{{disabledTransition ? 'Enabled' : 'Disabled'}} transition</m-button>
            </div>
            <p v-if="transitionEmit">Last transition emit: {{transitionEmit}}</p>
        </div>`
    }))
    .add('automatic-scroll=false', () => ({
        data: () => ({
            selectedTabIndex: 1,
            direction: MSlideTransitionDirection.LeftToRight,
            nbParagraphe: 20,
            nbTab: 3,
            skinNavbar: MNavbarSkin.TabLight
        }),
        methods: {
            showTab(newIndex): void {
                let _this: any = this as any;
                _this.direction = (newIndex < _this.selectedTabIndex) ? MSlideTransitionDirection.LeftToRight : MSlideTransitionDirection.RightToLeft;
                _this.selectedTabIndex = newIndex;
                _this.nbParagraphe = (_this.selectedTabIndex % 2) === 0 ? 30 : newIndex * 40;
            }
        },
        template: `
        <div>
            <m-navbar :selected="selectedTabIndex"
                      :skin="skinNavbar">
                <m-navbar-item v-for="index in nbTab"
                               :key="index"
                               :value="index"
                               @click="showTab(index)">
                    Tab {{index}}
                </m-navbar-item>
            </m-navbar>
            <m-slide-transition :direction="direction"
                                :automatic-scroll="false">
                <div v-for="index in nbTab"
                     :key="index"
                     v-if="selectedTabIndex === index"
                     class="m-u--margin-top">
                    <h3 class="m-u--no-margin">Tab {{index}}</h3>
                    <p v-for="index in nbParagraphe"
                       :key="index">Lorem ipsum dolor sit amet, consectetur adipisicing elit. A laudantium odio ipsum, quae quos cum dolor, omnis perferendis, veritatis deleniti iusto consectetur? Impedit tempora quam ab laborum maiores sapiente earum?</p>
                </div>
            </m-slide-transition>
        </div>`
    }));
