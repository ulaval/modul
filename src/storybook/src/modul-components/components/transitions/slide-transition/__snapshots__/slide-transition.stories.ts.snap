// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-slide-transition Tabs example 1`] = `
<div>
  <div
    class="m-navbar m--is-skin-tab-light m--is-multiline"
  >
    <div
      class="m-navbar__max-width"
      style="max-width: 1200px;"
    >
      <!---->
       
      <div
        class="m-navbar__contents"
      >
        <div
          class="m-navbar__wrap"
        >
          <ul
            class="m-navbar__list"
            role="menubar"
          >
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 1
                
                </span>
              </div>
            </li>
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 2
                
                </span>
              </div>
            </li>
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 3
                
                </span>
              </div>
            </li>
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 4
                
                </span>
              </div>
            </li>
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 5
                
                </span>
              </div>
            </li>
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 6
                
                </span>
              </div>
            </li>
            <li
              class="m-navbar-item"
              role="menuitem"
            >
              <div
                class="m-navbar-item__contents"
                role="button"
                style="white-space: nowrap;"
                tabindex="0"
              >
                <span
                  class="m-navbar-item__text"
                >
                  
                    Tab 7
                
                </span>
              </div>
            </li>
             
            <!---->
             
            <!---->
          </ul>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-slide-transition m--is-left-to-right-transition"
  >
    <div
      class="m-u--margin-top"
    >
      <h3
        class="m-u--no-margin"
      >
        Tab 1
      </h3>
       
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. A laudantium odio ipsum, quae quos cum dolor, omnis perferendis, veritatis deleniti iusto consectetur? Impedit tempora quam ab laborum maiores sapiente earum?
      </p>
    </div>
  </div>
   
  <div
    class="m-u--margin-top--m"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Add tab 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <button
      class="m-button m-u--margin-left m--is-skin-secondary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Delete tab 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <button
      class="m-button m-u--margin-left m--is-skin-secondary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Disabled transition 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
  </div>
   
  <!---->
</div>
`;
